pkg/scd/operational_intents_handler.go

```diff
@@ -16,6 +16,37 @@ import (
 	"github.com/interuss/stacktrace"
 )
 
+// subscriptionIsImplicitAndOnlyAttachedToOIR will check if:
+// - the subscription is defined and is implicit
+// - the subscription is attached to the specified operational intent
+// - the subscription is not attached to any other operational intent
+//
+// This is to be used in contexts where an implicit subscription may need to be cleaned up: if true is returned,
+// the subscription can be safely removed after the operational intent is deleted or attached to another subscription.
+//
+// NOTE: this should eventually be pushed down to CRDB as part of the queries being executed in the callers of this method.
+//
+//	See https://github.com/interuss/dss/issues/1059 for more details
+func subscriptionIsImplicitAndOnlyAttachedToOIR(ctx context.Context, r repos.Repository, oirID *dssmodels.ID, subscription *scdmodels.Subscription) (bool, error) {
+	if subscription == nil {
+		return false, nil
```

Comment:
* Subscriptions are allowed to be `nil` when the state of an OIR is 'accepted' so receiving a nil here does not necessarily indicate a problem
* when an OIR is created there is no previous subscription to speak of, so a nil makes sense as well

My intent of having the check here was probably to avoid the extra checks in the caller, but conciseness consideration taken aside:

I may have originally called the method `subscriptionNeedsCleanup`: at that time the early return with `false` would fit pretty well with the method name. But even now, if a subscription does not exist it feels Ok to just return false, as a non-existing subscription can't be implicit or attached to the OIR?

Reply:
Looking at the context where this method is called, I feel like moving the check out of it will not bring any benefits: I'd rather leave it as-is, especially knowing that an upcoming PR is going to remove this code.

Reply:
> if a subscription does not exist it feels Ok to just return false, as a non-existing subscription can't be implicit or attached to the OIR?

I guess that's a possible interpretation, but then I'd say that the root issue here is that the abstraction defined by this function is a bit awkward. Not a huge issue in itself here though.
As long as we are not silencing errors I'm OK, and looks like it's not the case here.

> especially knowing that an upcoming PR is going to remove this code

Is this actually planned for?
(but not a valid reason anyway)

Reply:
Yes, in https://github.com/interuss/dss/issues/1059 and it's part of the work we'd like to possibly get done until the 14th